addons/mail/static/src/utils/common/markdown.js

```diff
@@ -0,0 +1,73 @@
+// Extracted from the marked library
+function cleanUrl(href) {
+    try {
+        href = encodeURI(href).replace(/%25/g, '%');
+    }
+    catch {
+        return null;
+    }
+    return href;
+}
+
+let odooMarked;
+export function markdown(markdown) {
+    if (!odooMarked) {
+        const odooRenderer = {
+            useNewRenderer: true,
+            breaks: true,
+            renderer: {
+                // Override the link renderer to add `target="_blank"`
+                link({ href, title, tokens }) {
+                    const text = this.parser.parseInline(tokens);
+                    const cleanHref = cleanUrl(href);
+                    if (cleanHref === null) {
+                        return text;
+                    }
+                    href = cleanHref;
+                    let out = '<a href="' + href + '"';
```

Comment:
please don’t construct html manually, too easy to become an injection vector

Reply:
> Therefore it should be safe unless the marked lib isn't

But why take the risk and not use safer dom elements? Also, I see your override of the code block did not kept the `escape` that is [in the orginial code](https://github.com/markedjs/marked/blob/14dcd33d39a92ca48f7d0df68d6a546498fca432/src/Renderer.ts#L31) which in not really reasuring to prevent injections.

Reply:
Ho yes, because it's redundant with our own escaping

Reply:
We should write a test for each possibly problematic code from https://book.hacktricks.xyz/pentesting-web/xss-cross-site-scripting/xss-in-markdown (that was linked before), to ensure they are all covered